packages/relayer-lib/src/tx_builder/eth_to_cosmos.rs

```diff
@@ -457,46 +492,53 @@ where
             ..Default::default()
         };
 
+        let latest_signature_slot = headers.last().map(|h| h.consensus_update.signature_slot);
+
         // Final check to make sure the target chain's calculated slot is greater than our latest
-        // update slot:
+        // update's signature slot
         wait_for_condition(
             Duration::from_secs(15 * 60),
             Duration::from_secs(10),
             || async {
+                if headers.is_empty() {
+                    return Ok(true);
+                }
+
                 let latests_tm_block = self.tm_client.latest_block().await?;
                 let latest_onchain_timestamp = latests_tm_block.block.header.time.unix_timestamp();
                 let calculated_slot = ethereum_client_state
                     .compute_slot_at_timestamp(latest_onchain_timestamp.try_into().unwrap())
                     .unwrap();
                 tracing::debug!(
                     "Waiting for target chain to catch up to slot {}",
-                    latest_trusted_slot
+                    calculated_slot
                 );
-                Ok(calculated_slot > finality_update.signature_slot)
+                Ok(calculated_slot > latest_signature_slot.unwrap())
```

Comment:
In general, we want to avoid `unwrap` in the relayer since it will shut down the entire relayer api

Reply:
Agree with the other ones, but this one is safe because we only get here when this one is something

Reply:
Is there no way to make this type safe?

Reply:
Not really, since we don't have the consensus update anymore :)